File: Domain/Match.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Domain
{
    public class Match
    {
        public int Id { get; set; }

        public Team TeamA { get; set; }
        public int TeamAId { get; set; }

        public Team TeamB { get; set; }
        public int TeamBId { get; set; }

        public string EventName { get; set; }
        public int Series { get; set; }

        public short GameId { get; set; }
        public Game Game { get; set; }


        public ICollection<Prediction> Predictions { get; set; }
        public ICollection<MatchComment> MatchComments{ get; set; }

    }
}

File: client-side/src/app/models/match.ts
import { IPrediction, IPredictionStatus } from "./prediction";
import { ITeam } from "./team";

export interface IMatchEnvelope {
    matches: IMatch[];
    matchCount: number;
}

export interface IMatch {
    id: number;
    category: string;
    eventName: string;
    startDate: Date;
    settledDate: Date;
    teamA: ITeam;
    teamB: ITeam;
    winner: ITeam;
    predictions: IPrediction[];
    series: number;
    matchStatus: IPredictionStatus;
    game: IGame;
    comments: IComment[];
}

export interface IComment {
    id: string;
    displayName: string;
    message: string;
    photo: string;
    createdAt: Date;
}

export interface IGame {
    id: number;
    name: string;
    displayText: string;
}

export interface IMatchForm {
    eventName: string;
    teamAId: number;
    teamBId: number;
    gameId: number;
    series: number;
    title: string;
    description: string;
    startsAt: Date;
}

export interface IMatchRecent {
    teamA: ITeam;
    teamB: ITeam;
    winner: ITeam;
    settledDate: Date;
    id: number;
}

export interface IMatchPredictionRecent {
    userPhoto: string;
    predictionName: string;
    amount: number;
    when: Date;
}